src/pages/index.js

```diff
@@ -51,45 +104,155 @@ const addNewCardPopup = new PopupWithForm(
+  api
+    .updateProfileAvatar(inputValues.avatar)
+    .then(() => {
+      userInfo.setAvatar(inputValues.avatar);
+      avatarEditPopup.close();
+    })
+    .catch((err) => {
+      console.log(err);
+    })
+    .finally(() => {
+      avatarEditPopup.renderLoading(false, "Save");
+    });
+}
 
 function handleProfileFormSubmit(inputValues) {
-  const { title, description } = inputValues;
-  userInfo.setUserInfo(title, description);
-  profileEditPopup.close();
+  profileEditPopup.renderLoading(true);
+  api
+    .updateProfileInfo(inputValues)
+    .then(() => {
+      userInfo.setUserInfo(inputValues);
+      profileEditPopup.close();
+    })
+    .catch((err) => {
+      console.log(err);
+    })
+    .finally(() => {
+      profileEditPopup.renderLoading(false, "Save");
+    });
 }
 
 function handleNewCardSubmit(inputValues) {
-  const { title, url } = inputValues;
-  const newCardData = renderCard({ name: title, link: url });
-  section.addItem(newCardData);
-  addNewCardPopup.close();
+  addNewCardPopup.renderLoading(true);
+  api
+    .addNewCard(inputValues)
+    .then((cardData) => {
+      const newCardData = renderCard(cardData);
+      newCardSection.addItem(newCardData);
+      addNewCardPopup.close();
+    })
+    .catch((err) => {
+      console.log(err);
+    })
+    .finally(() => {
+      addNewCardPopup.renderLoading(false, "Create");
+    });
 }
 
-function handleCardClick(caption, imageUrl) {
-  cardPreviewPopup.open(caption, imageUrl);
+function handleCardClick(cardData) {
+  cardPreviewPopup.open(cardData);
 }
 
 function renderCard(cardData) {
-  const card = new Card(cardData, "#card-template", handleCardClick);
+  const card = new Card(
+    cardData.name,
+    cardData.link,
+    cardData.isLiked,
+    cardData.likes,
+    cardData._id,
+    userId,
+    cardSelector,
+    handleCardClick,
+    handleDeleteClick,
+    handleLikeClick
+  );
   return card.getView();
 }
 
+function handleDeleteClick(card) {
+  deleteCardPopup.open();
+  deleteCardPopup.setSubmitAction(() => {
+    deleteCardPopup.renderLoading(true);
+    api
+      .deleteCard(card.cardId)
+      .then(() => {
+        card.handleDeleteCard();
+        deleteCardPopup.close();
+      })
+      .catch((err) => {
+        console.log(err);
+      })
+      .finally(() => {
+        deleteCardPopup.renderLoading(false);
+      });
+  });
+}
+
+function handleProfileEditClick() {
+  const user = userInfo.getUserInfo();
+  profileTitleInput.value = user.name;
+  profileDescriptionInput.value = user.about;
+  profileEditPopup.open();
+}
+
+function handleLikeClick(card) {
+  if (card.isLiked) {
+    api
+      .removeCardLikes(card.cardId)
+      .then((res) => {
+        card.updateLikes(res.isLiked);
+      })
+      .catch((err) => {
+        console.log(err);
+      });
+  } else {
+    api
+      .addCardLikes(card.cardId)
+      .then((res) => {
+        card.updateLikes(res.isLiked);
+      })
+      .catch((err) => {
+        console.log(err);
```

Comment:
Good job that you `catch` possible errors at the end of server requests. You can simplify it with `.catch(console.error);` because `console.error` is meant to log errors